lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/IntegrationStrategy.cxx

```diff
@@ -92,6 +92,47 @@ String IntegrationStrategy::__repr__() const
 }
 
 
+struct IntegrationStrategyCoefficientsPolicy
+{
+  const UnsignedInteger marginalIndex_;
+  const NumericalSample & outputSample_;
+  const NumericalPoint & weights_;
+  const Matrix & designMatrix_;
+  const Indices & addedRanks_;
+  NumericalPoint & alpha_;
+
+  IntegrationStrategyCoefficientsPolicy(const UnsignedInteger marginalIndex,
+					const NumericalSample & outputSample,
+					const NumericalPoint & weights,
+					const Matrix & designMatrix,
+					const Indices & addedRanks,
+					NumericalPoint & alpha)
+    : marginalIndex_(marginalIndex)
+    , outputSample_(outputSample)
+    , weights_(weights)
+    , designMatrix_(designMatrix)
+    , addedRanks_(addedRanks)
+    , alpha_(alpha)
+  {
+    // Nothing to do
+  }
+
+  inline void operator()( const TBB::BlockedRange<UnsignedInteger> & r ) const
+  {
+    for (UnsignedInteger j = r.begin(); j != r.end(); ++j)
+      {
+	for (UnsignedInteger i = 0; i < outputSample_.getSize(); ++i)
+	  {
+	    const UnsignedInteger indexAdded(addedRanks_[j]);
+	    // The integral is approximated by a weighted sum of f * Phi
+	    alpha_[j] += weights_[i] * outputSample_[i][marginalIndex_] * designMatrix_(i, indexAdded);
```

Comment:
@regislebrun 
> For the algorithm, only the weighted version of the output sample is needed, but the user may want to access to the output sample, so both the weighted and unweighted versions of the output sample are needed.

You indeed have to keep the unweighted sample, but you only need a weighted marginal, not the whole weighted sample, so this is affordable.
In `IntegrationStrategy::computeCoefficients`, create a NP weightedMarginal of dimension `outputSample_.getSize()`, fill it with
```
weightedMarginal[i] = weights_[i] * outputSample_[i][marginalIndex_];
```
(no idea yet how to do that efficiently) and replace the first 3 arguments of IntegrationStrategyCoefficientsPolicy constructor by this NP.